tsl/src/continuous_aggs/Readme.md

```diff
@@ -1,144 +1,115 @@
+the same or be governed by manual refreshes.
+
+## Bookkeeping and Internal State ##
+
+TimescaleDB does bookkeeping for each continuous aggregate to know
+which buckets of the aggregates require refreshing. Whenever a
+modification happens to the source data, an invalidation for the
+modified region is written to an invalidation log. However,
+invalidations are not written after the *invalidation threshold*,
+which tracks the latest bucket materialized thus far. This threshold
+allows write amplification to be kept to a minimum by, e.g, never
+refreshing the most recent time bucket that is still being "filled".
+
+Thus, to store, maintain, and query aggregations, continuous
+aggregates consist of the following objects:
+
+1. A user view, which queries and finalizes the aggregations and is
+   also the object that users interact with.
+2. A partial view, which is used to materialize new data.
+3. A direct view, which holds the original query that users specified.
+4. An internal materialization hypertable, containing the materialized
+   data as partial aggregates for each time bucket.
+5. An invalidation threshold, which tracks how "far" the
+   materialization has reached and below which invalidations must be
+   logged.
+6. A trigger on the source hypertable that invalidates regions of data
+   in every transaction, based on INSERT, UPDATE, and DELETE
+   statements. The trigger writes its invalidations to the hypertable
+   invalidation log.
+7. A hypertable invalidation log that tracks invalidated regions of
+   data for each hypertable. Entries in this log contain time ranges
+   that need to be re-materialized across all the hypertable's
+   continuous aggregates.
+8. A materialization invalidation log. Once a refresh runs on a given
+   continuous aggregate, this log tracks how invalidations from the
+   hypertable invalidation log are processed against the refresh
+   window for the refreshed continuous aggregate. Thus, a single
+   invalidation in the hypertable invalidation log becomes one entry
+   per continuous aggregate in the materialization invalidation log.
+
+## The materialized hypertable ##
+
+The materialized hypertable does not store the aggregate's output, but
+rather the partial aggregate state. For instance, in case of an
+average, each bucket stores the sum and count in an internal binary
+form. The partial aggregates are what gives continuous aggregates
+flexibility; buckets can be individually updated and multiple partial
+aggregates can be combined to form new partials. Future enhancements
+may allow aggregating at different time resolutions using the the same
+underlying continuous aggregate.
+
+## The Invalidation Log and Threshold ##
+
+Mutating transactions must record their mutations in the invalidation
+log, so that a refresh knows to re-materialize the invalidated
+range. This happens by installing a trigger on the source hypertable
+when the first continuous aggregate on that hypertable is created.
```

Comment:
```suggestion
Mutating transactions must record their mutations in the invalidation
log for the hypertable, so that a refresh knows to re-materialize the invalidated
range. This happens by installing a trigger on the source hypertable
when the first continuous aggregate on that hypertable is created.
```

Reply:
Again, your suggestion could be interpreted as if there is one log per hypertable, which is not the case.